packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -0,0 +1,229 @@
+    rebase: (change, base, rebaseChild) => rebaseChild(change, base),
+};
+
+const singleNodeHandler: FieldChangeHandler<FieldChangeMap> = {
+    rebaser: singleNodeRebaser,
+    encoder: singleNodeEncoder,
+
+    intoDelta: (change, deltaFromChild) => [{
+        type: Delta.MarkType.Modify,
+        fields: deltaFromChild(change),
+    }],
+};
+
+const singleNodeField = new FieldKind(
+    brand("SingleNode"),
+    Multiplicity.Value,
+    singleNodeHandler,
+    (a, b) => false,
+    new Set(),
+);
+
+const fieldKinds: ReadonlyMap<FieldKindIdentifier, FieldKind> = new Map(
+    [singleNodeField, valueField].map((field) => [field.identifier, field]),
+);
+
+const family = new ModularChangeFamily(fieldKinds);
+
+const valueChange1a: ValueChangeset = { old: 0, new: 1 };
+const valueChange1b: ValueChangeset = { old: 0, new: 2 };
+const valueChange2: ValueChangeset = { old: 1, new: 2 };
+
+const innerChanges1a: FieldChangeMap = {
+    a: {
+        fieldKind: valueField.identifier,
+        change: brand(valueChange1a),
+    },
+};
+
+const innerChanges1b: FieldChangeMap = {
+    a: {
+        fieldKind: valueField.identifier,
+        change: brand(valueChange1b),
+    },
+
+    b: {
+        fieldKind: valueField.identifier,
+        change: brand(valueChange1a),
+    },
+};
+
+const innerChanges2: FieldChangeMap = {
+    a: {
+        fieldKind: valueField.identifier,
+        change: brand(valueChange2),
+    },
+
+    b: {
+        fieldKind: valueField.identifier,
+        change: brand(valueChange1a),
+    },
+};
+
+const rootChange1a: FieldChangeMap = {
+    a: {
+        fieldKind: singleNodeField.identifier,
+        change: brand(innerChanges1a),
+    },
+
+    b: {
+        fieldKind: valueField.identifier,
+        change: brand(valueChange2),
+    },
+};
+
+const rootChange1b: FieldChangeMap = {
+    a: {
+        fieldKind: singleNodeField.identifier,
+        change: brand(innerChanges1b),
+    },
+};
+
+const rootChange2: FieldChangeMap = {
+    a: {
+        fieldKind: singleNodeField.identifier,
+        change: brand(innerChanges2),
+    },
+};
+
+describe("ModularChangeFamily", () => {
+    it("compose", () => {
+        const composedValues: ValueChangeset = { old: 0, new: 2 };
+
+        const innerComposed: FieldChangeMap = {
+            a: {
+                fieldKind: valueField.identifier,
+                change: brand(composedValues),
+            },
+            b: {
+                fieldKind: valueField.identifier,
+                change: brand(valueChange1a),
+            },
+        };
+
+        const expectedCompose: FieldChangeMap = {
+            a: {
+                fieldKind: singleNodeField.identifier,
+                change: brand(innerComposed),
+            },
+            b: {
+                fieldKind: valueField.identifier,
+                change: brand(valueChange2),
+            },
+        };
+
+        assert.deepEqual(family.compose([rootChange1a, rootChange2]), expectedCompose);
```

Comment:
It would be nice to test that the value changes get composed too. (E.g., going from 0 to 2 in the value field)